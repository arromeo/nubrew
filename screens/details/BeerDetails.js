// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../../dev_port.json');

import React from 'react';
import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import VoteComponent from '../vote/VoteComponent.js';


export default class BeerDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
      voteCast: 'None',
      favorited: false,
    }
  }

  componentDidMount() {
    let url = `${port.DEV_PORT}/api/user/${this.props.user_id}/favorites`
    fetch(url)
      .then(res => res.json())
      .then(data => {
        let found = false;
        data['result'].forEach(beer => {
          if (beer['beer_id'] === this.props.data[0].beer_id) {
            found = true;
          }
        })
        this.setState({
          favorited: found,
        })
      })
      .catch(err => {
        console.log(err)
      });
  }

  render() {
    const updateVote = (vote, beer_id, user_id) => {
      return fetch(`${port.DEV_PORT}/api/user/:user_id/beer/:beer_id/vote`, 
        {
          method: "POST",
          headers: { "Content-Type": "application/json" },
          body: JSON.stringify({
            user_id, beer_id, vote
        }),
      })
      .catch((error) => {
        console.error(error);
      })
    }

    const addToFavoriteList = (user_id, beer_id) => {
      return fetch(`${port.DEV_PORT}/api/user/:user_id/beer/:beer_id/favorite`, 
        {
          method: "POST",
          headers: { "Content-Type": "application/json" },
          body: JSON.stringify({
            user_id, beer_id
        }),
      })
      .then(res => res.json())
      .then(data => {
        this.props.updateFavorites();
        // console.log(this.props.updateFavorites());
        this.setState({
          favorited: data.favorited
        })
      })
      .catch((error) => {
        console.error(error);
      })
    }

    const voteIndicator = (event, user_id, beer_id) => {
      let vote = 0;
      if (event > 0.5) {
        vote = 1;
        this.setState({
          voteCast: 'Liked',
        })
        updateVote(vote, beer_id, user_id);
      } else if (event < -0.5) {
        vote = -1;
        this.setState({
          voteCast: 'Disliked',
        })
        updateVote(vote, beer_id, user_id);
      } else {
        if (this.state.value === 0) {
          this.setState({
            value: 0.01,
          })
        } else {
          this.setState({
            value: 0,
          })
        }
      }
    }

    const beer = this.props.data[0];

    return (
      <View style={styles.container}>
        <View style={styles.contentContainer}>
          <Image source={{uri: beer.img_url}} style={{height: 200, width: 150}}/>
          <View style={styles.verticalContainer}>
            <TouchableOpacity 
              style={styles.buttonStyle}
              onPress={() => {
                this.props.navigate({
                  routeName: 'Find',
                });
              }}>
              <Ionicons style={styles.buttonIcon} name="md-search" size={25} color="#FFBC02"/>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.buttonStyle}
              onPress={() => {
                addToFavoriteList(this.props.user_id, beer.beer_id);
              }}>
              {this.state.favorited && 
                <View>
                  <Ionicons style={styles.buttonIcon} name="md-star" size={25} color="#FFBC02"/>
                </View>
              }
              {!this.state.favorited && 
                <View>
                  <Ionicons style={styles.buttonIcon} name="md-star" size={25} color="grey"/>
                </View>
              }
            </TouchableOpacity>
          </View>
        </View>
        <View style={styles.verticalContainer}>
          <Text style={styles.headerFont}>{beer.brewery_name}'s {beer.beer_name}</Text>
          <Text>Style: {beer.category}</Text>
          <View style={styles.verticalContainer}>
            <Text>ABV: {beer.abv}</Text>
            <Text>IBU: {beer.ibu}</Text>
          </View>
          <Text>{beer.beer_description}</Text>
        </View>
        <VoteComponent onSlidingComplete={voteIndicator} navigationParams={this.props.navigationParams} value={this.state.value} voteCast={this.state.voteCast} user_id={this.props.user_id}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  headerFont: {
    color: 'black',
    fontWeight: 'bold',
    flex: 1,
    textAlign: 'center',
    justifyContent: 'center',
  },
  verticalContainer: {
    flex: 0.5,
    margin: 10,
    justifyContent: "center",
    alignItems: 'center',
  },
  horizontalContainer: {
    flex: 0.5,
    paddingBottom: 5,
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: 'center',
  },
  contentContainer: {
    flex: 1,
    paddingBottom: 5,
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: 'center',
  },
  buttonStyle: {
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    flex: 0.2,
    width: '75%',
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#61170E',
  },
});
