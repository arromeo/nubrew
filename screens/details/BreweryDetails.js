import React from 'react';
import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class BreweryDetails extends React.Component {
  render() {
    const brewery = this.props.data[0];
    return (
      <View style={styles.container}>
        <Image style={{height: 200, width:300}} source={{uri: brewery.img_url}} resizeMode='contain'/>
        <Text style={styles.detailsTitle}>{brewery.name}</Text>
        <View style={styles.addressCard}>
          <Text>{brewery.street_address}</Text>
          <Text>{brewery.city}, {brewery.province}</Text>
          <Text>{brewery.postal_code}</Text>
        </View>
        <Text style={styles.description}>{brewery.description}</Text>
        <TouchableOpacity 
          style={styles.buttonStyle}
          onPress={() => {
            this.props.navigate({
              routeName: 'Map',
            });
          }}>
          <Ionicons style={styles.buttonIcon} name="md-pin" size={25} color="#FFBC02"/>
        </TouchableOpacity>
      </View>
    )
  }

}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    backgroundColor: '#FFFFFF',
    padding: 15
  },
  detailsTitle: {
    fontSize: 20,
    fontWeight: "bold",
  },
  addressCard: {
    marginLeft: 15,
    marginTop: 15,
    alignItems: 'center'
  },
  description: {
    marginTop: 15
  },
  buttonStyle: {
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    flex: 0.2,
    width: '75%',
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#61170E',
  },
});