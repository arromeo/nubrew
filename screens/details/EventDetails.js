import React from 'react';
import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class EventDetails extends React.Component {
  render() {
    const event = this.props.data[0];
    return (
      <View style={styles.container}>
        <Image style={{height: 200, width:300}} source={{uri: event.img_url}} resizeMode='contain'/>
        <Text style={styles.detailsTitle}>{event.name}</Text>
        <Text style={styles.description}>{event.store_name}</Text>
        <Text style={styles.description}>{event.details}</Text>
        <TouchableOpacity 
          style={styles.buttonStyle}
          onPress={() => {
            this.props.navigate({
              routeName: 'Map',
            });
          }}>
          <Ionicons style={styles.buttonIcon} name="md-pin" size={25} color="#FFBC02"/>
        </TouchableOpacity>
      </View>
    )
  }

}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    backgroundColor: '#FFFFFF',
    padding: 15
  },
  detailsTitle: {
    fontSize: 20,
    fontWeight: "bold",
  },
  addressCard: {
    marginLeft: 15,
    marginTop: 15,
    alignItems: 'center'
  },
  description: {
    marginTop: 15
  },
  buttonStyle: {
    marginTop: 10,
    marginBottom: 10,
    padding: 5,
    flex: 0.2,
    width: '75%',
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#61170E',
  },
});