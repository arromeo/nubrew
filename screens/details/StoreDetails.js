import React from 'react';
import { TouchableOpacity, View, Text, FlatList } from 'react-native';

export default class StoreDetails extends React.Component {
  render() {
    const styles = this.props.styles;
    return (
        <FlatList
        data={this.props.data}
        keyExtractor={item => item.id.toString()}
        renderItem={({item}) => 
          <TouchableOpacity style={styles.listItemContainer}
            onPress={() => {
              this.props.navigate('Detail', {
                category: "Store",
                id: item.id,
              })}
            }
          >
            <View style={styles.searchResultContainer}>
              <Text>{item.name}</Text>
              <Text>{item.description}</Text>
              <Text>{item.street_address}, {item.city}, {item.province}, {item.postal_code}</Text>
            </View>
          </TouchableOpacity>
        }
        />
    )
  }
}
