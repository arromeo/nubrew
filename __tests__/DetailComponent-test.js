import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import '../__mock__/xhr-mock';

import BeerDetails from '../screens/details/BeerDetails';
import BreweryDetails from '../screens/details/BreweryDetails';
import EventDetails from '../screens/details/EventDetails';
import StoreDetails from '../screens/details/StoreDetails';

import props from '../__mock__/props';

describe('Detail Components render properly', () => {
  it('BeerDetails component renders', () => {
    const tree = renderer.create(<BeerDetails data={props.beerData} navigate={props.navigation.navigate} user_id={props.user_id} navigationParams={props.navigationParams}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('BreweryDetails component renders', () => {
    const tree = renderer.create(<BreweryDetails data={props.breweryData} navigate={props.navigation.navigate}/>)
    expect(tree).toMatchSnapshot();
  })

  it('EventDetails component renders', () => {
    const tree = renderer.create(<EventDetails data={props.eventData} navigate={props.navigation.navigate}/>)
    expect(tree).toMatchSnapshot();
  })

  it('StoreDetails component renders', () => {
    const tree = renderer.create(<StoreDetails data={props.storeData} navigate={props.navigation.navigate}/>)
    expect(tree).toMatchSnapshot();
  })
})