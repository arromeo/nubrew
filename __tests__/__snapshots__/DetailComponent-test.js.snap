// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Components render properly BeerDetails component renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 150,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#61170E",
            "borderRadius": 25,
            "flex": 0.2,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "padding": 5,
            "width": "75%",
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#61170E",
            "borderRadius": 25,
            "flex": 0.2,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "padding": 5,
            "width": "75%",
          }
        }
      >
        <View>
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "flex": 1,
          "fontWeight": "bold",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      's 
    </Text>
    <Text>
      Style: 
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <Text>
        ABV: 
      </Text>
      <Text>
        IBU: 
      </Text>
    </View>
    <Text />
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingBottom": 20,
          "paddingLeft": 60,
          "paddingRight": 60,
          "paddingTop": 20,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightgrey",
            "borderRadius": 25,
            "flex": 0.2,
            "justifyContent": "center",
            "opacity": 0.8,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightgrey",
            "borderRadius": 25,
            "flex": 0.2,
            "justifyContent": "center",
            "opacity": 0.8,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;

exports[`Detail Components render properly BreweryDetails component renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text />
    <Text>
      , 
    </Text>
    <Text />
  </View>
  <Text
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 25,
        "flex": 0.2,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "padding": 5,
        "width": "75%",
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 25,
        "flex": 0.2,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "padding": 5,
        "width": "75%",
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`Detail Components render properly EventDetails component renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text />
    <Text>
      , 
    </Text>
    <Text />
  </View>
  <Text
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`Detail Components render properly StoreDetails component renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <Text />
    <Text>
      , 
    </Text>
    <Text />
  </View>
  <Text>
    
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 25,
        "flex": 0.2,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "padding": 5,
        "width": "75%",
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 25,
        "flex": 0.2,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "padding": 5,
        "width": "75%",
      }
    }
  >
    <Text />
  </View>
</View>
`;
